packages/angular/src/plugins/plugin.ts

```diff
@@ -0,0 +1,733 @@
+  return existsSync(cachePath) ? readJsonFile(cachePath) : {};
+}
+
+function writeProjectsToCache(
+  cachePath: string,
+  results: Record<string, AngularProjects>
+) {
+  writeJsonFile(cachePath, results);
+}
+
+export const createNodesV2: CreateNodesV2<AngularPluginOptions> = [
+  '**/angular.json',
+  async (configFiles, options, context) => {
+    const optionsHash = hashObject(options);
+    const cachePath = join(
+      workspaceDataDirectory,
+      `angular-${optionsHash}.hash`
+    );
+    const projectsCache = readProjectsCache(cachePath);
+    try {
+      return await createNodesFromFiles(
+        (configFile, options, context) =>
+          createNodesInternal(configFile, options, context, projectsCache),
+        configFiles,
+        options,
+        context
+      );
+    } finally {
+      writeProjectsToCache(cachePath, projectsCache);
+    }
+  },
+];
+
+async function createNodesInternal(
+  configFilePath: string,
+  options: AngularPluginOptions,
+  context: CreateNodesContextV2,
+  projectsCache: Record<string, AngularProjects>
+): Promise<CreateNodesResult> {
+  const angularWorkspaceRoot = dirname(configFilePath);
+
+  // Do not create a project if package.json isn't there
+  const siblingFiles = readdirSync(
+    join(context.workspaceRoot, angularWorkspaceRoot)
+  );
+  if (!siblingFiles.includes('package.json')) {
+    return {};
+  }
+
+  const hash = await calculateHashForCreateNodes(
+    angularWorkspaceRoot,
+    options,
+    context,
+    [getLockFileName(detectPackageManager(context.workspaceRoot))]
+  );
+
+  projectsCache[hash] ??= await buildAngularProjects(
+    configFilePath,
+    angularWorkspaceRoot,
+    options,
+    context
+  );
+
+  return { projects: projectsCache[hash] };
+}
+
+async function buildAngularProjects(
+  configFilePath: string,
+  angularWorkspaceRoot: string,
+  options: AngularPluginOptions,
+  context: CreateNodesContextV2
+): Promise<AngularProjects> {
+  const projects: Record<string, AngularProjects[string] & { root: string }> =
+    {};
+
+  const absoluteConfigFilePath = join(context.workspaceRoot, configFilePath);
+  const angularJson = readJsonFile<AngularJson>(absoluteConfigFilePath);
+
+  const appShellTargets: Target[] = [];
+  const prerenderTargets: Target[] = [];
+  for (const [projectName, project] of Object.entries(
+    angularJson.projects ?? {}
+  )) {
+    const targets: Record<string, TargetConfiguration> = {};
+
+    const projectTargets = getAngularJsonProjectTargets(project);
+    if (!projectTargets) {
+      continue;
+    }
+
+    const namedInputs = getNamedInputs(project.root, context);
+
+    for (const [targetName, angularTarget] of Object.entries(projectTargets)) {
+      const externalDependencies = ['@angular/cli'];
+
+      targets[targetName] = {
+        command: `ng run ${projectName}:${targetName}`,
+        options: { cwd: angularWorkspaceRoot },
+        metadata: {
+          technologies: ['angular'],
```

Comment:
@FrozenPandaz should I also add `webpack` and `esbuild` to the build targets depending on the executor used?

Reply:
We discussed this offline and decided not to.